layer/VkLayer_FROG_gamescope_wsi.cpp

```diff
@@ -525,6 +590,24 @@ namespace GamescopeWSILayer {
       GetPhysicalDeviceFeatures2(pDispatch, physicalDevice, pFeatures);
     }
 
+    static VkResult GetPhysicalDeviceSurfacePresentModesKHR(
+      const vkroots::VkInstanceDispatch* pDispatch,
+        VkPhysicalDevice                 physicalDevice,
+        VkSurfaceKHR                     surface,
+        uint32_t*                        pPresentModeCount,
+        VkPresentModeKHR*                pPresentModes) {
+      static constexpr std::array<VkPresentModeKHR, 1> s_FifoPresentModes = {{
+        VK_PRESENT_MODE_FIFO_KHR,
+      }};
+
+      if (auto state = GamescopeSurface::get(surface)) {
+        if (gamescopeIsForcingFifo() && state->frameLimiterAware())
+          return vkroots::helpers::array(s_FifoPresentModes, pPresentModeCount, pPresentModes);
+      }
+
+      return pDispatch->GetPhysicalDeviceSurfacePresentModesKHR(physicalDevice, surface, pPresentModeCount, pPresentModes);
```

Comment:
Need to patch GetSurfaceCapabilities2 as well. The present mode group needs to be limited to FIFO as well.